libmediation/types/types.cpp

```diff
@@ -21,6 +22,19 @@
 
 using namespace std;
 
+regex invalidChars()
+{
+#ifndef _WIN32
+    // / and ASCII 0 to 31
+    regex invalid("[/\x00-\x1F]");
```

Comment:
...you change the `invalid` objects to be `static const`, which means they only get initialised once : there's no need to create a new object every time this function is entered. You will also be able to pass `std::regex_constants::ECMAScript | std::regex_constants::optimize` as the second constructor parameter, potentially instructing the regex engine to construct an object that matches faster at the expense of longer construction time.
I'm not aware of any engines that actually use the `optimize` thing yet, but, well, you know...